### ISSUE TITLE
Hint when a developer uses await in a method that doesn't have async

### ISSUE THREAD

**Issue by sethladd** [dart-lang/sdk#22625]
When a developer uses code like this:

myMethod() {
&nbsp;&nbsp;var x = await myAsyncMethod();
}

Our runtimes tell the user that they are missing a semicolon. In almost all cases, the real error is that the user forgot to put the word async in the method signature.

A hint to the developer would be a really big productivity improvement.

**Comment by sethladd** [comment]
dart2js prints a warning if it encounters an await without an async. Very cool, and helpful for our developers.  https://codereview.chromium.org/1094903002/diff/80001/pkg/compiler/lib/src/warnings.dart

```
~/tmp $ dart2js bug.dart
bug.dart:4:13:
Error: Expected ';' after this.
  var foo = await awesome();
            ^^^^^
bug.dart:4:13:
Warning: Cannot resolve 'await'.
Did you mean to add the 'async' marker to 'main'?
  var foo = await awesome();
            ^^^^^
Hint: When run on the command-line, the compiled output might require a preamble file located in:
  <sdk>/lib/_internal/compiler/js_lib/preambles.
Error: Compilation failed.
```

**Comment by srawlins** [comment]
Would a HINT be accepted into the analyzer for this? It could be a linter lint, but I think the benefits of reaching a developer by default (opt-out analyzer, rather than opt-in linter) are huge. Huge huge. @sethladd notes that dart2js prints a warning. And I like Emasoft's argument:

> no source code loaded into the Dart Editor or passed through the Analyzer should keep that keyword as a variable name

**Comment by bwilkerson** [comment]
While I agree with the sentiment, I'd probably make it a lint for now. If we ever get to make breaking changes to the language, then making these true keywords is something we should strongly consider.

**Comment by Emasoft** [comment]
@sethladd I find very difficult to follow or reply to the discussions. Is there any way for me to take back the ownership of my posts?

**Comment by srawlins** [comment]
How about a comprimise? We can just alter the text of an UNDEFINED_IDENTIFIER error, so its just like dart2js: https://codereview.chromium.org/2156003004/

**Comment by srawlins** [comment]
Fixed in e60c1e4c595aaa264a261314dbd3a7d49a5aa232.
